src/CommonRuntime/Caching.fs

```diff
@@ -35,16 +36,19 @@ let createInMemoryCache expiration =
 open System.Collections.Concurrent
 
 /// Creates a cache that uses in-memory collection
-let createInMemoryCache expiration = 
+let createInMemoryCache (expiration:TimeSpan) = 
   let dict = new ConcurrentDictionary<_, _>()
   { new ICache<_> with
       member __.Set(key, value) = 
-        dict.[key] <- (value, DateTime.UtcNow)
+        dict.AddOrUpdate(key, (value, DateTime.UtcNow), (fun _ _ -> value, DateTime.UtcNow)) |> ignore
+        async { do! Async.Sleep (expiration.Milliseconds)
+                if dict <> null then
+                    dict.TryRemove(key) |> ignore } |> Async.Start
```

Comment:
This async block was missing and will fix a memory leek. Can you do do the same on the other implementation above that doesn't use ConcurrentDictionary?